### PR TITLE
New JS test: draw shaderball with parquet wood texture.

### PR DESCRIPTION
**Pull request by prideout**
Shameless recreation of the Android `textured-object` sample. I will make this into a tutorial soon.

Also made some improvements to the test-builder Python script. Really, this script exists only to make iteration easier; we need a better solution for automated tests...

screen shot 2018-10-25 at 12 35 25 pm

### CODE REVIEW

File: libs/filamentjs/tests/test_parquet.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    const sampler = new Filament.TextureSampler(
+        Filament.MinFilter.LINEAR_MIPMAP_LINEAR,
+        Filament.MagFilter.LINEAR,
+        Filament.WrapMode.CLAMP_TO_EDGE);
+
+    const ao = Filament.createTextureFromPng(Filament.assets['floor_ao_roughness_metallic.png'], engine);
+    const basecolor = Filament.createTextureFromPng(Filament.assets['floor_basecolor.png'], engine);
+    const normal = Filament.createTextureFromPng(Filament.assets['floor_normal.png'], engine);
+    matinstance.setTextureParameter("aoRoughnessMetallic", ao, sampler)
```
Comment (bejado) [inline libs/filamentjs/tests/test_parquet.js]:
```suggestion
    matinstance.setTextureParameter('aoRoughnessMetallic', ao, sampler)
```

File: libs/filamentjs/tests/test_parquet.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    matinstance.setTextureParameter("baseColor", basecolor, sampler)
```
Comment (bejado) [inline libs/filamentjs/tests/test_parquet.js]:
```suggestion
    matinstance.setTextureParameter('baseColor', basecolor, sampler)
```

File: libs/filamentjs/tests/test_parquet.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    matinstance.setTextureParameter("normal", normal, sampler)
```
Comment (bejado) [inline libs/filamentjs/tests/test_parquet.js]:
```suggestion
    matinstance.setTextureParameter('normal', normal, sampler)
```

File: libs/filamentjs/tests/test_parquet.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    this.swapChain = engine.createSwapChain();
+    this.renderer = engine.createRenderer();
+    this.camera = engine.createCamera();
+    this.view = engine.createView();
+    this.view.setCamera(this.camera);
+    this.view.setScene(this.scene);
+    this.resize();
+    this.render = this.render.bind(this);
+    this.resize = this.resize.bind(this);
+    window.addEventListener("resize", this.resize);
```
Comment (bejado) [inline libs/filamentjs/tests/test_parquet.js]:
```suggestion
    window.addEventListener('resize', this.resize);
```

**Comment by romainguy** [comment]
Your colors seem off. It looks like the texture is not samples as srgb maybe? It should be darker.

**Comment by prideout** [comment]
Ah right, I need to add an SRGB option to the flags that get passed to `createTextureFromPng`.  Let me do that in a subsequent PR.

**Comment by prideout** [comment]
Gah, I should have squashed those commits (GitHub's "apply suggestion" feature), apologies for polluting our commit history!

**Pull request merged**